File: Julia/Arima/arimaGDPC1.jl
# https://lost-stats.github.io/Time_Series/ARIMA-models.html
# https://lampspuc.github.io/StateSpaceModels.jl/latest/manual/

# Load necessary packages
using StateSpaceModels, CSV, Dates, DataFrames, LinearAlgebra, Plots

# Import data
data = CSV.File("Dataset/Arima/GDPC1.csv") |> DataFrame

# Specify GDPC1 series as an ARIMA(2,2,2) model
model = SARIMA(data.GDPC1, order = (2,2,2))

# Fit (estimate) the model
fit!(model)

# Print estimates
print_results(model)

forec = forecast(model, 24)
plot(model, forec; legend = :topleft, title="ARIMA Prediction", label=["Original Series" "Prediction"], lindewidth=3)

File: Python/Arima/arimaGDPC1.py
# https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.sarimax.SARIMAX.html
import pandas as pd
import statsmodels.api as sm
import matplotlib.pyplot as plt

# Import (download) data
data = pd.read_csv("../../Dataset/Arima/GDPC1.csv")

# Specify GDPC1 series as an ARIMA(2,2,2) model
model = sm.tsa.SARIMAX(data['GDPC1'], order=(2, 2, 2))

# Fit (estimate) the model
results = model.fit()

# Print estimates
print(results.summary())

# Forecast the next 24 values
forecast_values = results.get_forecast(steps=24)

# Plotting the results
plt.figure(figsize=(10, 6))
plt.plot(data['GDPC1'], label='Original Series', linewidth=1)
plt.plot(forecast_values.predicted_mean, label='Prediction', linewidth=1, color='red')
plt.title('ARIMA Prediction')
plt.legend(loc='upper left')
plt.show()
